modules/ipcscript.js

```diff
@@ -48,6 +50,21 @@ function IPCScript_getMetaStr() {
   return this._metaStr;
 });
 
+IPCScript.prototype.dedup = function() {
+  // scripts sent via e10s message manager contain new string instances every time a new one is created -> dedup them to reduce memory footprint
+
+  // alternative approaches:
+  // a) beam down IPC script instances only once via the parent process manager every time they change.
+  // b) creating resource URIs and loading user scripts straight from URIs might also be more elegant
+  //    see: https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/Using#Programmatically_adding_aliases
+  var old = instances.get(this.uuid);
+
+  if(old == this.textContent)
+    this.textContent = old
+  else
+    instances.set(this.uuid, this.textContent)
```

Comment:
So before this change, the script string is in memory only while the script is executing/is in scope.  It is garbage collected once it goes out of scope (i.e. navigation to a different page, close the tab, or as soon as it finishes executing if it doesn't expose references to itself to content).

After this change, every script that has ever executed has a copy of its source stored permanently in memory.  When there are several children processes (there is only one today AIUI) there will be several copies, and none of them will ever go away until the browser is completely closed.

It's not clear to me whether this new approach is better overall in terms of memory usage.